### PR TITLE
[DISCUSS]Add new interfaces that can support Schema

### PR DESCRIPTION
**Pull request by duhenglucky**
Add new interfaces that can support Schema

### CODE REVIEW

File: openmessaging-api/src/main/java/io/openmessaging/api/schema/GenericRecord.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @version OMS 2.0.0
+ * @since OMS 2.0.0
+ */
+public interface GenericRecord {
+    /**
+     * Return the schema version of this record.
+     * @return Schema version
+     */
+    int getSchemaVersion();
```
Comment (dbl-x) [inline openmessaging-api/src/main/java/io/openmessaging/api/schema/GenericRecord.java:32]: why schema version, not return schema id?

File: openmessaging-api-samples/src/main/java/io/openmessaging/samples/consumer/SchemaConsumerApp.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+public class SchemaConsumerApp {
+
+    public static void main(String[] args) {
+        //Load and start the vendor implementation from a specific OMS driver URL.
+        final MessagingAccessPoint messagingAccessPoint =
+            OMS.builder()
+                .region("Shanghai")
+                .endpoint("127.0.0.1:9876")
+                .driver("rocketmq")
```
Comment (vongosling) [inline openmessaging-api-samples/src/main/java/io/openmessaging/samples/consumer/SchemaConsumerApp.java:37]: Driver name could be an enum.

File: openmessaging-api/src/main/java/io/openmessaging/api/OMSConsumer.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.openmessaging.api;
+
+public interface OMSConsumer {
```
Comment (vongosling) [inline openmessaging-api/src/main/java/io/openmessaging/api/OMSConsumer.java:19]: The interface should have Javadoc.

File: openmessaging-api/src/main/java/io/openmessaging/api/schema/GenericRecord.java (hunk 1 above)
Comment (vongosling) [reply openmessaging-api/src/main/java/io/openmessaging/api/schema/GenericRecord.java:32]: why naming to Record?

**Pull request closed without merging**